### PR TITLE
Update 3Di driver to expose groundwater and surface water as individual meshes

### PR DESCRIPTION
**Pull request by uclaros**
On the 3Di driver, in addition to the existing `Mesh2D` mesh, the driver now allows to open `Mesh2D_groundwater` and `Mesh2D_surface_water` individual meshes

Also fixes dataset min/max statistics which returned nan/nan, breaking the symbology in QGIS

### CODE REVIEW

File: tests/test_3di.cpp [hunk 1]
```diff
@@ -13,11 +13,454 @@
 #include "mdal_testutils.hpp"
 #include "mdal_utils.hpp"
 
+TEST( Mesh3DiTest, Mesh2Dgroundwater )
+{
+  std::string path = test_file( "/3di/2d_groundwater/results_3di.nc" );
+  EXPECT_EQ( MDAL_MeshNames( path.c_str() ), "3Di:\"" + path + "\":Mesh2D;;3Di:\"" + path + "\":Mesh2D_groundwater;;3Di:\"" + path + "\":Mesh2D_surface_water" );
+  std::string uri = "3Di:\"" + path + "\":Mesh2D_groundwater";
+//  std::string uri = "3Di:\"" + path + "\":Mesh2D";
```
Comment (PeterPetrik) [inline tests/test_3di.cpp:21]: remove

File: tests/test_3di.cpp [hunk 2]
```diff
@@ -13,11 +13,454 @@
...
+
+  MDAL_CloseMesh( m );
+}
+
+TEST( Mesh3DiTest, Mesh2Dsurface_water )
+{
+  std::string path = test_file( "/3di/2d_groundwater/results_3di.nc" );
+  EXPECT_EQ( MDAL_MeshNames( path.c_str() ), "3Di:\"" + path + "\":Mesh2D;;3Di:\"" + path + "\":Mesh2D_groundwater;;3Di:\"" + path + "\":Mesh2D_surface_water" );
+  std::string uri = "3Di:\"" + path + "\":Mesh2D_surface_water";
+  //  std::string uri = "3Di:\"" + path + "\":Mesh2D";
```
Comment (PeterPetrik) [inline tests/test_3di.cpp:240]: remove

File: mdal/frmts/mdal_3di.hpp [hunk 3]
```diff
@@ -16,6 +16,31 @@
...
+
+      virtual size_t scalarData( size_t indexStart, size_t count, double *buffer ) override;
+      virtual size_t vectorData( size_t indexStart, size_t count, double *buffer ) override;
+
+    private:
+      std::vector< size_t > mRequestedMeshFaceIds;
+  };
 
   /**
    * Driver of 3Di file format.
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_3di.hpp:46]: can you add few lines about the 3 meshes we support

**Review by PeterPetrik — APPROVED** [review]
Looks good to me.

Please 
- (optional) add few lines here: https://github.com/lutraconsulting/MDAL/blob/master/docs/source/drivers/3di.rst
- bump the MDAL version 
- update https://github.com/lutraconsulting/MDAL?tab=readme-ov-file#versioning-and-integration-in-qgis
- create PR in QGIS

**Comment by uclaros** [comment]
Thanks for the review @PeterPetrik .
There's another PR to be made after this is merged and prior to bumping the version.

**Pull request merged**